### PR TITLE
Add Cosine and Correlation Distance Support to Scikit-learn-intelex

### PR DESCRIPTION
**Pull request by richardnorth3**
## Description

Adds Python bindings for the cosine and correlation distance algorithms in scikit-learn-intelex, enabling users to compute cosine and correlation based pairwise distances with oneDAL.

---
**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] I have run it locally and tested the changes extensively.
- [ ] All CI jobs are green or I have provided justification why they aren't.
- [ ] I have extended testing suite if new functionality was introduced in this PR.

### CODE REVIEW

**Comment by david-cortes-intel** [comment]
I think this is supposed to first go through preview, like covariance currently is, but waiting for comments from @icfaust about whether there'd be any issue with putting it there.

**Comment by icfaust** [comment]
> I think this is supposed to first go through preview, like covariance currently is, but waiting for comments from @icfaust about whether there'd be any issue with putting it there.

You are absolutely right, though so far the PR does not include any sklearnex public interfaces to the added onedal pybind11 pairwise distances calculations.

**Comment by david-cortes-intel** [comment]
> > I think this is supposed to first go through preview, like covariance currently is, but waiting for comments from @icfaust about whether there'd be any issue with putting it there.
> 
> You are absolutely right, though so far the PR does not include any sklearnex public interfaces to the added onedal pybind11 pairwise distances calculations.

But it does include them as a metric type for nearest neighbors.

**Comment by icfaust** [comment]
@richardnorth3 what is the status of this development?
